### PR TITLE
Remove split checkout toggle and legacy checkout

### LINKED ISSUE #10636: Remove split_checkout toggle and legacy checkout
## Description

All instances have split checkout activated. We are ready to remove the toggle and legacy checkout.

## Acceptance Criteria & Tests

1. Check there is no toggle on split checkout in `/admin/feature-toggle/features`
2. Test checkout as guest with Cash, Stripe and Paypal
3. Test checkout with account  with Cash, Stripe and Paypal

### PR DESCRIPTION
**Pull request by abdellani**
#### What? Why?

- Closes #10636

Removes `split_checkout` feature toggle and updates various parts of the test suite that were not correctly using the new checkout and order flow logic.

Useful references:
https://github.com/openfoodfoundation/openfoodnetwork/pull/8810

#### What should we test?

99% of these changes are removing old checkout code split_checkout feature toggles and updating parts of the test suite that were not using the new checkout logic. Aside from that there's a couple of small changes here to actual application code in https://github.com/openfoodfoundation/openfoodnetwork/pull/10913/commits/32b8d7201af4effc892f330848984d1f1db1ed05 and https://github.com/openfoodfoundation/openfoodnetwork/pull/10913/commits/4a8a1468d1c944a27ac796aa2441675fb8591ae2 which are effectively fixing bugs that are already in production but were not being picked up in the test suite...

I've pulled one of those changes out into a separate PR here: #10933 which could be tested separately (if that's easier). It affects marking the order as `complete` after processing stripe payments that require additional authorization (3DS).

The other change mostly affects Paypal redirects after a payment has failed for some reason at the checkout (like insufficient funds). It should now redirect to the checkout with the correct error message (currently I think it throws a 404/not found).

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/models/spree/order.rb
Comment (abdellani) [inline app/models/spree/order.rb:324]: why do we return when split_checkout is enabled?

**Comment by abdellani** [comment]
`rspec ./spec/services/process_payment_intent_spec.rb:110 `
`rspec ./spec/services/process_payment_intent_spec.rb:130`

are failing because the expected order state is 'complete' and now, it's 'confimation'.
I tried to make a payment using x-4242, and it's working without any issues (obviously because the confirm code doesn't use ProcessPaymentIntent).  
`PaymentGateways::StripeController#authorize`, I need to check where in the UI, this endpoint is used.

`rspec ./spec/controllers/payment_gateways/paypal_controller_spec.rb:91`
`rspec ./spec/controllers/payment_gateways/paypal_controller_spec.rb:102`
are related to  Paypal, I'll need to do some research to see how can I run Paypal in dev mode.

Multiple tests are failing due to [this PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/8810)
`rspec ./spec/system/consumer/shopping/cart_spec.rb:184` 
After , the checkout page doesn't show taxes if the split checkout is enabled.
There are other tests:
`rspec ./spec/models/spree/adjustment_spec.rb:251`
`rspec ./spec/models/spree/adjustment_spec.rb:275`
...
before continuing on the other tests, I prefer to get a feedback about `create_tax_charge!`  because many tests depend on it.
@Matt-Yorkley  can you please tell me what do you think?

File: app/helpers/terms_and_conditions_helper.rb
Comment (Matt-Yorkley) [inline app/helpers/terms_and_conditions_helper.rb:27]: I think the `platform_terms_required?` and `distributor_terms_required?` methods are still in use..?

**Comment by Matt-Yorkley** [comment]
Looks good so far :+1:

Yeah in this case I think the specs mostly need to be changed to reflect the what the code is doing and not the other way around. For example, taxes should not be applied before the order gets towards the end of the checkout process (`payment` state, I think) and there are some technical reasons for this (in some cases taxes need to be different depending on the user's address, but they haven't necessarily saved their address before the `address` state is passed on the `/checkout/details` step). 

So those tests need updating a bit and I'd say that test at `spec/system/consumer/shopping/cart_spec.rb:184` can just be deleted.

In those tax adjustment specs around `spec/models/spree/adjustment_spec.rb:251` I think some of the test setup might need to be modified a bit so that the order being tested has correctly passed to the `payment` state with a valid `shipping_method`, `payment_method` and `address`, and maybe the current tests are not doing that and operating on an order in `cart` state?

For reference, I had a little draft PR here that was aimed at some of the cleanup related to this and the question of when taxes should (or shouldn't) be applied: https://github.com/openfoodfoundation/openfoodnetwork/pull/10776/files

**Review by Matt-Yorkley — APPROVED** [review]
Great :heart:

I think we should get this merged ASAP :+1:

**Review by dacook — CHANGES_REQUESTED** [review]
Great work, this turned out to be a big task!

It's clear there were still a lot of tests based on legacy checkout, thanks for working through these. 
I have a few questions, and a suggestion for tidying up `state_machine_spec.rb`.

But I'm concerned about removing the concurrency spec, and disabling a number of others. Supporting the checkout process is pretty important and ideally we'd like to protect against bugs that we previous had specs for. Perhaps a practical compromise is to postpone these, but I think that needs to be agreed by product-circle or scheduled as a task.

File: app/models/spree/order/checkout.rb
Comment (abdellani) [reply app/models/spree/order/checkout.rb:80]: if we keep the original code  `before_transition`, `create_tax_charge!` will do nothing because of the guard clause.
```ruby
    def create_tax_charge!
      return if state.in?(["cart", "address", "delivery"])
...
    end
```
It was introduced on this [PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/8810) for optimization reasons.

File: spec/requests/checkout/concurrency_spec.rb
Comment (abdellani) [reply spec/requests/checkout/concurrency_spec.rb:1]: maybe I'm wrong. From what I understand, this test is related to the removed controller `CheckoutController`
```diff
-    allow_any_instance_of(CheckoutController).
-      to receive(:checkout_workflow).
-      and_wrap_original do |method, *args|
-
-      breakpoint_reached_counter += 1
-      breakpoint.synchronize {}
-      method.call(*args)
-    end
```

File: app/models/spree/order/checkout.rb
Comment (Matt-Yorkley) [reply app/models/spree/order/checkout.rb:80]: Taxing the items in the order needs to happen after the order's addresses have been set but should be done before the user lands on the payments step page. It wasn't quite working correctly before as the order's state hadn't been updated at the point it was being checked in the `before_transition` hook. The `after_transition` hook is the correct place for it :+1:

File: spec/requests/checkout/failed_checkout_spec.rb
Comment (Matt-Yorkley) [reply spec/requests/checkout/failed_checkout_spec.rb:47]: All of these tests are designed to only work with the old checkout, so they currently don't work when the legacy checkout code is removed. We can rewrite them at some point (hence the pending flag) but the main issue being addressed here is that split checkout has already been rolled out and is in production, but 99% of the test suite is not using that new logic, which includes changes to things like order workflow that effect critical code paths in various places.

File: spec/models/spree/order/state_machine_spec.rb
Comment (dacook) [reply spec/models/spree/order/state_machine_spec.rb:28]: That's a good idea too 👍

File: app/models/spree/order/checkout.rb
Comment (dacook) [reply app/models/spree/order/checkout.rb:80]: Thanks. It would be cool if we could add a spec to show that, but the main thing is we're improving 👍

File: spec/models/spree/order/checkout_spec.rb
Comment (dacook) [reply spec/models/spree/order/checkout_spec.rb]: 🙈

File: spec/requests/checkout/concurrency_spec.rb
Comment (dacook) [reply spec/requests/checkout/concurrency_spec.rb:1]: Thanks, what I meant is that the principle of testing concurrency is still important. That is, we're testing there is no possible race condition if you double-click the checkout button.

Of course, this test would need to be updated to suit the new split checkout controller, but I think we should. I'll suggest it in a new issue.

File: spec/requests/checkout/failed_checkout_spec.rb
Comment (dacook) [reply spec/requests/checkout/failed_checkout_spec.rb:47]: Thanks, I see now this is a pre-existing issue, and this is a massive improvement and shouldn't be blocked. I've created a new issue for the remaining specs: [Re-enable and update checkout specs](https://github.com/openfoodfoundation/openfoodnetwork/issues/10958)

File: spec/requests/checkout/concurrency_spec.rb
Comment (dacook) [reply spec/requests/checkout/concurrency_spec.rb:1]: New issue: [Re-enable and update checkout specs](https://github.com/openfoodfoundation/openfoodnetwork/issues/10958)

File: spec/models/spree/order/state_machine_spec.rb
Comment (dacook) [reply spec/models/spree/order/state_machine_spec.rb:26]: Done

**Review by dacook — APPROVED** [review]
Great teaming up on this massive much-needed cleanup!

Sorry for holding this up, I didn't see it clearly initially. I agree we need to move it forward.

Assuming you're happy with my little code cleanup, I'll mark ready for testing.

**Comment by Matt-Yorkley** [comment]
Rebased to resolve conflicts :ok_hand:

**Comment by abdellani** [comment]
I remember that I removed it locally. Maybe I accidentally overwritten the commit that updates the list when I fetched the changes from the repo.

Anyway, enabling or disabling that feature will not have any impact on the current behavior. 
I'll remove it on another PR.

**Pull request merged**
